### PR TITLE
fix(curriculum): Date Validation in Tests with Non-UTC Server Timezones 

### LINKED ISSUE #57381: Hardcoded Date Validation in Tests Fails with Non-UTC Server Timezones
### Describe the Issue

I encountered issues while attempting to pass the following tests in the "Exercise Tracker" project:

> **Test 8:** The response returned from `POST /api/users/:_id/exercises` will be the user object with the exercise fields added.

> **Test 15:** The `date` property of any object in the `log` array returned from `GET /api/users/:_id/logs` should be a string. Use the `dateString` format of the Date API.

These tests failed due to discrepancies between my local server's timezone and the hardcoded validation used in the tests. Specifically, the tests expect the `date` property to be in the `toDateString()` format (e.g., `"Mon Jan 01 1990"`) in UTC, but on a server with a local timezone (e.g., UTC-6), the `date` output reflects the local time, causing the validation to fail.

### Affected Page

https://www.freecodecamp.org/learn/back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker

### Your code

```js
res.json({
    _id: user._id,
    username: user.username,
    description: savedExercise.description,
    duration: savedExercise.duration,
    date: savedExercise.date.toDateString(),
});
```

### Expected behavior

The tests should pass when:
1. The server correctly handles the provided date (`1990-01-01`) and formats it using `toDateString()` or an equivalent.
2. The tests account for potential timezone differences in server configurations or explicitly require the developer to standardize the server to UTC.

### Screenshots

_No response_

### System

- Device: PC
- OS: Windows 11
- Browser: Edge
- Version: 131.0.2903.70 (Official build) (64-bit)

### Additional context

- My server's local timezone is UTC-6, which caused `new Date('1990-01-01').toDateString()` to output `"Sun Dec 31 1989"`, whereas the test expects `"Mon Jan 01 1990"`. This mismatch was caused by the hardcoded validation of the `date` field in the tests.
https://github.com/freeCodeCamp/freeCodeCamp/blob/3571a44a6eb9979fdf77b9b8893a44284525d99a/curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md?plain=1#L203-L223

- Below is an example payload and response highlighting the issue:

   **Payload**:
   ```bash
   description=test&duration=60&date=1990-01-01
   ```

   **Response**:
   ```json
   {
       "_id": "674d314bb40c32994064e9b1",
       "username": "fcc_test_17331121392",
       "description": "test",
       "duration": 60,
       "date": "Sun Dec 31 1989"
   }
   ```

   As shown, the server correctly processes the date and outputs it based on the local timezone, but this doesn't match the hardcoded UTC-based expectation of "Mon Jan 01 1990" in the test suite.

- To resolve the issue, I added `TZ=UTC` to my `.env` file to force the server to use UTC, which ensured that the output of `toDateString()` matched the test's expectations.

While this workaround was effective, it is not documented in the project instructions or requirements. Developers unfamiliar with timezone handling may spend unnecessary time debugging this issue.

---

**Suggestion:**

To improve the developer experience, I propose the following:
1. Add a **disclaimer** in the project description or test suite explaining that the server timezone must be set to UTC to ensure the tests pass.
2. Alternatively, update the tests to account for potential timezone differences. For example:
   - Validate the `date` field against both UTC and local `toDateString()` outputs, or
   - Require the server to explicitly return a UTC date string, such as `"Mon Jan 01 1990"`.

By implementing one of these changes, developers can avoid confusion and focus on building the project.

### PR DESCRIPTION
**Pull request by Abhishek-dev479**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #57381

### CODE REVIEW

File: curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md [hunk 1]
```diff
@@ -212,15 +212,22 @@ async (getUserInput) => {
...
+    const expectedDate = new Date(expected.date); // Jan 1, 1990
+    const allowedPreviousDate = new Date(expectedDate);
+    allowedPreviousDate.setDate(expectedDate.getDate() - 1); // Dec 31, 1989
+    const isValidDate =
+      receivedDate.toDateString() === expectedDate.toDateString() ||
+      receivedDate.toDateString() === allowedPreviousDate.toDateString();
+    assert.isTrue(
+      isValidDate,
+      `Expected date to be ${expectedDate.toDateString()} or ${allowedPreviousDate.toDateString()}, but got ${receivedDate.toDateString()}`
+    );
```
Comment (a2937) [inline curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md:230]: You have removed the tests for the username, __id, description and the duration for this section.

**Review by a2937 — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md (hunk 1 above)
Comment (Abhishek-dev479) [reply curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md:237]: I have added the tests again.

File: curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md [hunk 2]
```diff
@@ -495,26 +509,38 @@ async(getUserInput) => {
...
+    const expectedDates = [
+      new Date(currentDate.setDate(currentDate.getDate() - 1)).toLocaleDateString("en-US", {
+        timeZone: "UTC", weekday: "short", month: "short",
+        day: "2-digit", year: "numeric"
+      }).replaceAll(',', ''),
+      new Date().toLocaleDateString("en-US", {
+        timeZone: "UTC", weekday: "short", month: "short",
+        day: "2-digit", year: "numeric"
+      }).replaceAll(',', ''),
+      new Date(currentDate.setDate(currentDate.getDate() + 2)).toLocaleDateString("en-US", {
```
[the remark this answers is not in the document]
Comment (Abhishek-dev479) [reply curriculum/challenges/english/05-back-end-development-and-apis/back-end-development-and-apis-projects/exercise-tracker.md]: You are right. I have corrected it.

**Review by a2937 — APPROVED** [review]
Everything seems to be in working order.

**Comment by a2937** [comment]
While the date in the future is unnecessary, I don't see it breaking anything.

**Comment by Abhishek-dev479** [comment]
@ilenia-magoni can you assign another review.

**Comment by ilenia-magoni** [comment]
@Abhishek-dev479 no, a second reviewer will come depending on availability and expertise. @a2937 had specifically said she wanted to review the PR for the issue, assigning her was the way to give her notice the PR existed.

**Comment by Abhishek-dev479** [comment]
Alright.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**